pkg/proto/nbcontract/v1/customlinuxosconfig.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+package nbcontract.v1;
+
+message CustomLinuxOSConfig {
+    // The size in MB of a swap file that will be created on each node
+    optional int32 swap_file_size = 1;
+  
+    // Valid values are "always", "defer", "defer+madvise", "madvise" and "never"
+    optional string transparent_hugepage_support = 2;
+  
+    // Valid values are "always", "madvise" and "never"
+    optional string transparent_defrag = 3;
+  
+    // Sysctl settings for Linux agent nodes
+    optional SysctlConfig sysctl_config = 4;
+  
+    // Ulimit settings for Linux agent nodes
+    optional UlimitConfig ulimit_config = 5;
+  }
+
+message SysctlConfig {
```

Comment:
As discussed out-of-band, let's capture (maybe in the contract design doc?) the trade-offs between explicit fields for each value vs. more generic approach, for when we revisit this (and other relevant sets of fields.)